extensions/sftp/processors/ListSFTP.cpp

```diff
@@ -0,0 +1,1320 @@
+    if (latest_listed_entry_timestamp_this_cycle == last_listed_latest_entry_timestamp_) {
+      const auto& latest_files = ordered_files.at(latest_listed_entry_timestamp_this_cycle);
+      uint64_t elapsed_time = std::chrono::duration_cast<std::chrono::milliseconds>(current_run_time - last_run_time_).count();
+      /* If a precision-specific listing lag has not yet elapsed since out last execution, we wait. */
+      if (elapsed_time < listing_lag) {
+        logger_->log_debug("The latest listed entry timestamp is the same as the last listed entry timestamp (%lu) "
+                           "and the listing lag has not yet elapsed (%lu ms < % lu ms). Yielding.",
+                           latest_listed_entry_timestamp_this_cycle,
+                           elapsed_time,
+                           listing_lag);
+        context->yield();
+        return;
+      }
+      /*
+       * If we have already processed the entities with the newest timestamp,
+       * and there are no new entities with that timestamp, there is nothing to do.
+       */
+      if (latest_listed_entry_timestamp_this_cycle == last_processed_latest_entry_timestamp_ &&
+          std::all_of(latest_files.begin(), latest_files.end(), [this](const Child& child) {
+            return latest_identifiers_processed_.count(child.getPath()) == 1U;
+          })) {
+        logger_->log_debug("The latest listed entry timestamp is the same as the last listed entry timestamp (%lu) "
+                           "and all files for that timestamp has been processed. Yielding.", latest_listed_entry_timestamp_this_cycle);
+        context->yield();
+        return;
+      }
+    } else {
+      /* Determine the minimum reliable timestamp based on precision */
+      uint64_t minimum_reliable_timestamp = now * 1000 - listing_lag;
+      if (remote_system_timestamp_precision == TARGET_SYSTEM_TIMESTAMP_PRECISION_SECONDS) {
+        minimum_reliable_timestamp -= minimum_reliable_timestamp % 1000;
+      } else {
+        minimum_reliable_timestamp -= minimum_reliable_timestamp % 60000;
+      }
+      /* If the latest timestamp is not old enough, we wait another cycle */
+      if (minimum_reliable_timestamp < latest_listed_entry_timestamp_this_cycle) {
+        logger_->log_debug("Skipping files with latest timestamp because their modification date is not smaller than the minimum reliable timestamp: %lu ms >= %lu ms",
+            latest_listed_entry_timestamp_this_cycle,
+            minimum_reliable_timestamp);
+        ordered_files.erase(latest_listed_entry_timestamp_this_cycle);
+      }
+    }
+
+    for (auto& files_for_timestamp : ordered_files) {
+      if (files_for_timestamp.first == last_processed_latest_entry_timestamp_) {
+        /* Filter out previously processed entities. */
+        for (auto it = files_for_timestamp.second.begin(); it != files_for_timestamp.second.end();) {
```

Comment:
That is a really good point for a vector, so much so that I started to reimplement it with std::remove_if, then I realized that it is a list, not a vector, precisely because I wanted to erase from it in O(1).